src/components/organisms/UserProfileModal/UserProfileModal.tsx

```diff
@@ -26,48 +27,54 @@ import Button from "components/atoms/Button";
 import "./UserProfileModal.scss";
 
 type PropTypes = {
-  show: boolean;
-  onHide: () => void;
   zIndex?: number;
-  userProfile?: WithId<User>;
 };
 
-const UserProfileModal: React.FunctionComponent<PropTypes> = ({
-  show,
-  onHide,
+export const UserProfileModal: React.FunctionComponent<PropTypes> = ({
   zIndex,
-  userProfile,
 }) => {
   const venue = useSelector(currentVenueSelector);
 
   const { user } = useUser();
 
-  const chosenUserId = userProfile?.id;
-
   const { selectRecipientChat } = useChatSidebarControls();
 
+  const {
+    selectedUserProfile,
+    hasSelectedProfile,
+    closeUserProfileModal,
+  } = useProfileModalControls();
+
+  const chosenUserId = selectedUserProfile?.id;
+
   const openChosenUserChat = useCallback(() => {
     if (!chosenUserId) return;
 
     selectRecipientChat(chosenUserId);
     // NOTE: Hide the modal, after the chat is opened;
-    onHide();
-  }, [selectRecipientChat, onHide, chosenUserId]);
+    closeUserProfileModal();
+  }, [selectRecipientChat, closeUserProfileModal, chosenUserId]);
 
-  if (!userProfile || !chosenUserId || !user) {
+  if (!selectedUserProfile || !chosenUserId || !user) {
     return <></>;
   }
 
   // REVISIT: remove the hack to cast to any below
   return (
-    <Modal show={show} onHide={onHide} style={zIndex && { zIndex }}>
+    <Modal
+      show={hasSelectedProfile}
+      onHide={closeUserProfileModal}
+      style={zIndex && { zIndex }}
+    >
       <Modal.Body>
         <div className="modal-container modal-container_profile">
           <div className="profile-information-container">
             <div className="profile-basics">
               <div className="profile-pic">
                 <img
-                  src={userProfile.pictureUrl || "/default-profile-pic.png"}
+                  src={
+                    selectedUserProfile.pictureUrl || "/default-profile-pic.png"
+                  }
```

Comment:
`"/default-profile-pic.png"` should be moved to a const in `src/settings.ts` (as something like `DEFAULT_PROFILE_PIC`) and then referenced here.

(I feel like we already made this change somewhere.. maybe it's in one of @denisdimitrov's outstanding branches or similar?)